apps/camera_pipe/camera_pipe_generator.cpp

```diff
@@ -426,10 +447,52 @@ void CameraPipe::generate() {
             .estimate(x, 0, 2592)
             .estimate(y, 0, 1968);
 
+    } else if (get_target().has_gpu_feature()) {
+
+        // We can generate slightly better code if we know the output is even-sized
+        if (!auto_schedule) {
+            // TODO: The autoscheduler really ought to be able to
+            // accomodate bounds on the output Func.
+            Expr out_width = processed.width();
+            Expr out_height = processed.height();
+            processed.bound(c, 0, 3)
+                .bound(x, 0, (out_width / 2) * 2)
+                .bound(y, 0, (out_height / 2) * 2);
+        }
+
+        Var xi, yi, xii, xio;
+
+        /* 1391us on a gtx 980. */
+        processed.compute_root()
+            .reorder(c, x, y)
+            .unroll(x, 2)
+            .gpu_tile(x, y, xi, yi, 28, 12);
```

Comment:
Might be good to briefly comment where this tile size came from (so things fit in shared memory?).